### PR TITLE
NTRP-360 adding authorization with organizationId

### PR DESCRIPTION
**Pull request by paolomanca-pagopa**

### CODE REVIEW

File: packages/pull-signal/src/model/domain/errors.ts [hunk 1]
```diff
@@ -20,14 +20,14 @@ export function requestValidationError(message: string): ApiError<ErrorCodes> {
...
   eserviceId,
+  consumerId,
 }: {
-  purposeId: string;
+  consumerId: string;
   eserviceId: string;
 }): ApiError<ErrorCodes> {
   return new ApiError({
-    detail: `Insufficient privileges: cannot access to pull signals with voucher's purpose ${purposeId} for e-service PULL; please verify if the consumer has a valid agreement to pull signals to e-service ${eserviceId}`,
+    detail: `Insufficient privileges: cannot access to pull signals with voucher's organization ${consumerId} for e-service PULL. Please verify if the consumer has a valid agreement and a valid purpose to pull signals to e-service ${eserviceId}. Verify also if the e-service is enabled for using Signal Hub.`,
```
Comment (borgesis95) [inline packages/pull-signal/src/model/domain/errors.ts:30]: Should we align error message with push-service message? Should be: "I**nsufficient privileges: producerId: ${producerId} cannot access to pull signal for e-service with id ${eserviceId}; Please verify if you have a valid agreement and a valid purpose to pull signals. Verify also if that e-service is enabled for using Signal Hub.**"

File: packages/pull-signal/src/services/interop.service.ts [hunk 2]
```diff
@@ -4,33 +4,35 @@ import { interopRepository } from "../repositories/interop.repository.js";
...
       const result = await interopRepository(db).findBy(
         eserviceId,
-        purposeId,
+        consumerId,
+        eserviceState,
         purposeState,
         agreementState
       );
-      if (result === null) {
+      if (!result?.length) {
```
Comment (borgesis95) [inline packages/pull-signal/src/services/interop.service.ts:32]: should be move this into a method? just to read and understand what code do here

File: packages/pull-signal/test/interopService.test.ts [hunk 3]
```diff
@@ -2,77 +2,295 @@ import { beforeAll, describe, expect, it } from "vitest";
...
     await dataResetForSignalConsumers(postgresDB, config.interopSchema);
-    await dataPreparationForSignalConsumers(postgresDB, config.interopSchema);
   });
 
-  it("should give permission to a signals consumer to pull a signal", async () => {
-    const purposeId = authorizedPurposeIdForPullSignals;
-    const eserviceId = eserviceIdPushSignals;
+  it("Should deny permission to a signal consumer without e-service, agreement, purpose", async () => {
+    const consumerId = getUUID();
+    const eserviceId = getUUID();
```
Comment (borgesis95) [inline packages/pull-signal/test/interopService.test.ts:24]: I'll say: **"Should deny permission to a signal consumer without an agreement and purpose with an e-service"** . In that case e-service is not owned by consumer.

File: packages/pull-signal/test/interopService.test.ts [hunk 4]
```diff
@@ -2,77 +2,295 @@ import { beforeAll, describe, expect, it } from "vitest";
...
   });
 
-  it("Should deny permission to a signal consumer without purpose available", async () => {
-    const purposeId = "wrong-purpose-id";
-    const eserviceId = eserviceIdPushSignals;
+  it("Should deny permission to a signal consumer without e-service", async () => {
+    const consumerId = getUUID();
+    const eserviceId = getUUID();
+    const eservice = undefined;
+    const agreement = { eserviceId, consumerId };
```
Comment (borgesis95) [inline packages/pull-signal/test/interopService.test.ts:39]: I'll say: **"Should deny permission to a signal consumer for an unavailable e-service"**.

**Review by borgesis95 — CHANGES_REQUESTED** [review]

File: packages/pull-signal/src/model/domain/errors.ts (hunk 1 above)
Comment (paolomanca-pagopa) [reply packages/pull-signal/src/model/domain/errors.ts:30]: Got it, I'try to rewrite in this way:
`Insufficient privileges: your organization (id: ${consumerId}) cannot pull signals for an e-service with id ${eserviceId}. Please verify if you have a valid agreement and at least one valid purpose to pull signals. Verify also if that e-service is enabled for using Signal Hub.`

File: packages/pull-signal/src/services/interop.service.ts (hunk 2 above)
Comment (paolomanca-pagopa) [reply packages/pull-signal/src/services/interop.service.ts:32]: Got it! This Is a code with a "missed abstraction level".

**Review by borgesis95 — APPROVED** [review]

**Pull request merged**